src/App.tsx

```diff
@@ -1,44 +1,195 @@
+      setIsWrongPostResponse(false);
+    }
+
+    getPosts(selectedUserId)
+      .then(response => setPosts(response))
+      .catch(() => setIsWrongPostResponse(true))
+      .finally(() => setIsPostLoading(false));
+  };
+
+  const handleSidebarButton = (currentPost: Post) => {
+    setSelectedPost(current => {
+      return current === currentPost ? null : currentPost;
+    });
+    setIsCommentsLoading(true);
+    setIsCommentForm(false);
+
+    getComments(currentPost.id)
+      .then(response => setComments(response))
+      .catch(() => setIsWrongSidebarResponse(true))
+      .finally(() => setIsCommentsLoading(false));
+  };
+
+  const handleSubmitCommentForm = (
+    event: React.FormEvent<HTMLFormElement>,
+    reset: (inputType: Reset) => void,
+    name: string,
+    email: string,
+    body: string,
+    postId?: number,
+  ) => {
+    event.preventDefault();
+
+    if (!name || !email || !body) {
+      setIsDangerSubmit(true);
+
+      return;
+    }
+
+    setIsloadingNewComment(true);
+
+    const data = {
+      postId,
+      name,
+      email,
+      body,
+    };
+
+    const updateComments = (newComment: Comment) => {
+      setComments(currenList => {
+        return currenList
+          ? [
+            ...currenList,
+            newComment,
+          ]
+          : [newComment];
+      });
+
+      setIsDangerSubmit(false);
+      reset(Reset.comment);
+    };
+
+    setIsWarningUpdate(false);
+
+    postComment(data)
+      .then((response) => updateComments(response))
+      .catch(() => setIsWarningUpdate(true))
+      .finally(() => setIsloadingNewComment(false));
+  };
+
+  const canselDangerInput = () => {
+    setIsDangerSubmit(false);
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    const prevComments = comments;
+
+    setComments(currentList => {
+      if (currentList) {
+        const list = currentList?.filter(comment => comment.id !== commentId);
+
+        return list;
+      }
+
+      return currentList;
+    });
+
+    deleteComment(commentId)
+      .then()
+      .catch(() => setComments(prevComments));
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  userName={userName}
+                  isActiveList={isActiveList}
+                  onHandleUserSelect={handleUserSelect}
+                  onHandleSelectButton={handleSelectButton}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
+                {!userId && (
+                  <p data-cy="Nopost">
```

Comment:
```suggestion
                  <p data-cy="NoSelectedUser">
```
Keep all the original data-cy attributes to pass the tests
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/96409563/176f738c-829b-4612-b729-d682610d6608)